ebos/eclwriter.hh

```diff
@@ -254,33 +254,21 @@ public:
         if (reportStepNum == 0)
             return;
 
-        Scalar curTime = simulator_.time() + simulator_.timeStepSize();
-        Scalar totalCpuTime =
+        const Scalar curTime = simulator_.time() + simulator_.timeStepSize();
+        const Scalar totalCpuTime =
             simulator_.executionTimer().realTimeElapsed() +
             simulator_.setupTimer().realTimeElapsed() +
             simulator_.vanguard().externalSetupTime();
 
-        Opm::data::Wells localWellData = simulator_.problem().wellModel().wellData();
+        const auto localWellData  = simulator_.problem().wellModel().wellData();
+        const auto localGroupData = simulator_.problem().wellModel()
+            .groupData(reportStepNum, simulator_.vanguard().schedule());
 
-        Opm::data::Group localGroupData = simulator_.problem().wellModel().groupData(reportStepNum, simulator_.vanguard().schedule());
-
-        const auto& gridView = simulator_.vanguard().gridView();
-        int numElements = gridView.size(/*codim=*/0);
-        bool log = collectToIORank_.isIORank();
-        eclOutputModule_.allocBuffers(numElements, reportStepNum, isSubStep, log, /*isRestart*/ false);
-
-        ElementContext elemCtx(simulator_);
-        ElementIterator elemIt = gridView.template begin</*codim=*/0>();
-        const ElementIterator& elemEndIt = gridView.template end</*codim=*/0>();
-        for (; elemIt != elemEndIt; ++elemIt) {
-            const Element& elem = *elemIt;
-            elemCtx.updatePrimaryStencil(elem);
-            elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
-            eclOutputModule_.processElement(elemCtx);
-        }
+        this->prepareLocalCellData(isSubStep, reportStepNum);
```

Comment:
You can move this call above the creation of localWellData and localGroupData.

Reply:
> You can move this call above the creation of localWellData and localGroupData.

I'll do that the next time I'm touching this file.